wdpk/entware/cgi-bin/entware.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python3
+import cgitb
+
+cgitb.enable()
+
+import cgi
+import os
+import re
+import subprocess
+import sys
+import urllib.parse
+
+os.environ["PATH"] = os.path.pathsep.join(["/opt/bin", os.environ["PATH"]])
+REQUEST_METHOD = os.environ["REQUEST_METHOD"]
+QUERY_STRING = os.environ["QUERY_STRING"]
+
+
+def is_post():
+    return REQUEST_METHOD == "POST"
+
+
+def get_package_name(qs):
+    package_name = qs.get("package_name", [])
+    package_name = package_name[0] if package_name else ""
+
+    if not re.match(r"^\w+$", package_name):
```

Comment:
> CGI: To be on the safe side, if you must pass a string gotten from a form to a shell command, you should make sure the string contains only alphanumeric characters, dashes, underscores, and periods.

We might also want to provide a kill switch for this Entware manager as anyone on the internal network can remove packages without authentication. The kill switch would simply remove the cgi symlink.

Reply:
On my tests the cgi-bin at least on MyCloudEX2Ultra were reachable only if I was logged in